pkg/common/catalog/fill.go

```diff
@@ -218,51 +221,67 @@ func isInterfaceOrStructOfInterfaces(t reflect.Type) bool {
 	}
 }
 
-func parseTag(tag reflect.StructTag) (min, max int, err error) {
-	min = 0
-	max = 0
-	value := tag.Get("catalog")
-	if value == "" {
-		return min, max, nil
+type fieldOpts struct {
+	min    int
+	max    int
+	ignore bool
+}
+
+func parseFieldOpts(fieldTag reflect.StructTag) (_ fieldOpts, err error) {
+	catalogTag := fieldTag.Get("catalog")
+	if catalogTag == "" {
+		return fieldOpts{}, nil
 	}
 
+	var opts fieldOpts
 	maxset := false
-	for _, kv := range strings.Split(value, ",") {
-		parts := strings.SplitN(kv, "=", 2)
-		if len(parts) < 2 {
-			return 0, 0, fmt.Errorf("expected key=value for catalog tag value %q", kv)
+	for _, tagValue := range strings.Split(catalogTag, ",") {
+		parts := strings.SplitN(tagValue, "=", 2)
+		key := parts[0]
+		var value string
+		if len(parts) > 1 {
+			value = parts[1]
 		}
-		k := parts[0]
-		v := parts[1]
-		switch k {
+		switch key {
+		case "-":
```

Comment:
The meaning of "-" may be difficult to guess. Perhaps we can update the big comment of the Catalog interface explaining this?